### PR TITLE
[NO JIRA][BpkLoadingButton]: Updating TS types and removing propTypes ahead of React 18

### PR DESCRIPTION
**Pull request by olliecurtis**
As part of the migration to React 18 when running tests we found errors that were being through due to the mix of TS types and and propTypes which are deprecated and discouraged since React 17

```
console.error
    Warning: BpkLoadingButton: Support for defaultProps will be removed from function components in a future major release. Use JavaScript default parameters instead.
        at className (/Users/olivercurtis/Documents/Backpack/backpack/packages/bpk-component-loading-button/src/BpkLoadingButton.tsx:39:11)
```

Remember to include the following changes:

- [x] Ensure the PR title includes the name of the component you are changing so it's clear in the release notes for consumers of the changes in the version e.g `[KOA-123][BpkButton] Updating the colour`
- [ ] `README.md` (If you have created a new component)
- [ ] Component `README.md`
- [x] Tests
- [ ] Accessibility tests
    - The following checks were performed:
        - [ ] Ability to navigate using a [keyboard only](https://webaim.org/techniques/keyboard/)
        - [ ] Zoom functionality ([Deque University explanation](https://dequeuniversity.com/checklists/web/text)):
            - [ ] The page SHOULD be functional AND readable when only the text is magnified to 200% of its initial size
            - [ ] Pages must reflow as zoom increases up to 400% so that content continues to be presented in only one column i.e. Content MUST NOT require scrolling in two directions (both vertically and horizontally)
        - [ ] Ability to navigate using a [screen reader only](https://webaim.org/articles/screenreader_testing/)
- [ ] Storybook examples created/updated
- [ ] For breaking changes or deprecating components/properties, migration guides added to the description of the PR. If the guide has large changes, consider creating a new Markdown page inside the component's docs folder and link it here

### CODE REVIEW

File: packages/bpk-component-loading-button/src/BpkLoadingButton.tsx [hunk 1]
```diff
@@ -156,39 +156,4 @@ const BpkLoadingButton = (props: LoadingProps) => {
   );
 };
 
-BpkLoadingButton.propTypes = {
-  children: PropTypes.node.isRequired,
-  className: PropTypes.string,
```
Comment (amburrrshand) [inline packages/bpk-component-loading-button/src/BpkLoadingButton.tsx:161]: Should `classname` be included in the destructed props above?

Comment (olliecurtis) [reply packages/bpk-component-loading-button/src/BpkLoadingButton.tsx:161]: Thats a good catch! Having a deeper look, its not actually used in the component itself! So I am wondering if this was added for typing in the past but would just follow `...rest` and go down to the BpkButton.

AFAIK flow/propTypes didn't have the concept of merging props with that of other components 🤔

File: packages/bpk-component-loading-button/src/BpkLoadingButton-test.tsx [hunk 2]
```diff
@@ -16,9 +16,7 @@
  * limitations under the License.
  */
 
-/* @flow strict */
```
Comment (Sybsw) [inline packages/bpk-component-loading-button/src/BpkLoadingButton-test.tsx:19]: Hey Ollie could you please explain the removals of `flow strict` and the changings of `toMatchSnapshot` ?

Comment (olliecurtis) [reply packages/bpk-component-loading-button/src/BpkLoadingButton-test.tsx:19]: As this file uses Typescript having flow in here does nothing and flow isn't installed in the repo anymore for a long time.

In terms of the snapshots, moved away from these as they aren't reliable and to use a more reliable testing over snapshots, something we advocate for in Skyscanner

File: packages/bpk-component-loading-button/src/BpkLoadingButton.tsx [hunk 3]
```diff
@@ -79,33 +79,33 @@ type LoadingProps = {
...
-    large,
-    link,
-    linkOnDark,
-    loading,
-    primaryOnDark,
-    primaryOnLight,
-    secondary,
-    secondaryOnDark,
+    iconOnly = false,
+    iconPosition = ICON_POSITION.TRAILING,
```
Comment (amburrrshand) [inline packages/bpk-component-loading-button/src/BpkLoadingButton.tsx:100]:
```suggestion
    iconPosition = ICON_POSITION.TRAILING || ICON_POSITION. LEADING,,
```

File: packages/bpk-component-loading-button/src/BpkLoadingButton.tsx [hunk 4]
```diff
@@ -79,33 +79,33 @@ type LoadingProps = {
   large?: boolean,
   link?: boolean,
   linkOnDark?: boolean,
-  loading: boolean,
-  iconOnly: boolean,
+  loading?: boolean,
+  iconOnly?: boolean,
   icon?: ReactElement<any>,
-  iconPosition: string,
+  iconPosition?: string,
```
Comment (amburrrshand) [inline packages/bpk-component-loading-button/src/BpkLoadingButton.tsx:85]:
```suggestion
  iconPosition?: ICON_POSITION.TRAILING || ICON_POSITION. LEADING,
```

**Pull request merged**